src/main/java/com/ibm/crypto/plus/provider/ECDHKeyAgreement.java

```diff
@@ -312,7 +312,7 @@ protected void engineInit(Key key, AlgorithmParameterSpec params, SecureRandom r
             throw new InvalidKeyException("Key is not an ECPrivateKey");
         }
         java.security.interfaces.ECPrivateKey ecPrivKey = (java.security.interfaces.ECPrivateKey) key;
-
+        sun.security.util.ECUtil.checkPrivateKey(ecPrivKey);
```

Comment:
I checked this `ECUtil` class, the method `checkPrivateKey()` is not in JDK17 and JDK11. Do we need to or how we can check the invalid EC private key for JDK17 or JDK11?

Reply:
We need to manually implement the method for jdk versions later than 21...

Reply:
So, before 21, there will be another OpenJDK PR for this `checkPrivateKey()` implementation?

Reply:
we will basically use the sun implementation here.